algorithms/slovak.sbl

```diff
@@ -1,87 +1,229 @@
+groupings ( vowel )
 
-integers ( pV p1 )
+stringescapes {}
 
-groupings ( v )
+stringdef A"   '{U+00C1}'    // dlhe a
+stringdef A:   '{U+00C4}'    // siroke a
+stringdef C<   '{U+010C}'   // c s makcenom
+stringdef D<   '{U+010E}'   // d s makcenom
+stringdef E"   '{U+00C9}'    // e s dlznom
+stringdef E<   '{U+011A}'   // e s makcenom
+stringdef I"   '{U+00CD}'    // i s dlznom
+stringdef L"   '{U+0139}'   // l s dlznom
+stringdef L<   '{U+013E}'   // l s makcenom
+stringdef N<   '{U+013D}'   // n s makcenom
+stringdef O>   '{U+00D4}'    // o s vokanom
+stringdef O"   '{U+00D3}'    // o s dlznom
+stringdef R"   '{U+0154}'   // r s dlznom
+stringdef R<   '{U+0158}'   // r s makcenom
+stringdef S<   '{U+0160}'   // s s makcenom
+stringdef T<   '{U+0164}'   // t s makcenom
+stringdef U"   '{U+00DA}'    // u s dlznom
+stringdef U^   '{U+016E}'   // u s kruzkom
+stringdef Y"   '{U+00DD}'    // y s dlznom
+stringdef Z<   '{U+017D}'   // z s makcenom
 
-stringescapes {}
+stringdef a"   '{U+00E1}'    // dlhe a
+stringdef a:   '{U+00E4}'    // siroke a
+stringdef c<   '{U+010D}'   // c s makcenom
+stringdef d<   '{U+010F}'   // d s makcenom
+stringdef e"   '{U+00E9}'    // e s dlznom
+stringdef e<   '{U+011B}'   // e s makcenom
+stringdef i"   '{U+00ED}'    // i s dlznom
+stringdef l"   '{U+013A}'   // l s dlznom
+stringdef l<   '{U+013E}'   // l s makcenom
+stringdef n<   '{U+0148}'   // n s makcenom
+stringdef o>   '{U+00F4}'    // o s vokanom
+stringdef o"   '{U+00F3}'    // o s dlznom
+stringdef r"   '{U+0155}'   // r s dlznom
+stringdef r<   '{U+0159}'   // r s makcenom
+stringdef s<   '{U+0161}'   // s s makcenom
+stringdef t<   '{U+0165}'   // t s makcenom
+stringdef u"   '{U+00FA}'    // u s dlznom
+stringdef u^   '{U+016F}'   // u s kruzkom
+stringdef y"   '{U+00FD}'    // y s dlznom
+stringdef z<   '{U+017E}'   // z s makcenom
+stringdef ia   'ia'        // ia
+stringdef ie   'ie'        // ie
+stringdef iu   'iu'        // iu
 
-stringdef a' '{U+00E1}'
-stringdef a" '{U+00E4}'
-stringdef c^ '{U+010D}'
-stringdef d^ '{U+010F}'
-stringdef e' '{U+00E9}'
-stringdef i' '{U+00ED}'
-stringdef l' '{U+013A}'
-stringdef l^ '{U+013E}'
-stringdef n^ '{U+0148}'
-stringdef o' '{U+00F3}'
-stringdef o^ '{U+00F4}'
-stringdef r' '{U+0155}'
-stringdef s^ '{U+0161}'
-stringdef t^ '{U+0165}'
-stringdef u' '{U+00FA}'
-stringdef y' '{U+00FD}'
-stringdef z^ '{U+017E}'
-
-define v 'aeiouy{a'}{a"}{e'}{i'}{o'}{o^}{u'}{y'}'
-
-define mark_regions as (
-
-    $pV = limit
-    $p1 = limit
+define vowel 'aeiyou'
+
+define un_accent as repeat (
+	[substring] among (
+		'{a"}' (<-'a')
+		'{a:}' (<-'a')
+		'{c<}' (<-'c')
+		'{d<}' (<-'d')
+		'{e"}' (<-'e')
+		'{e<}' (<-'e')
+		'{i"}' (<-'i')
+		'{l"}' (<-'l')
+		'{l<}' (<-'l')
+		'{n<}' (<-'n')
+		'{o>}' (<-'o')
+		'{o"}' (<-'o')
+		'{r"}' (<-'r')
+		'{r<}' (<-'r')
+		'{s<}' (<-'s')
+		'{t<}' (<-'t')
+		'{u"}' (<-'u')
+		'{u^}' (<-'u')
+		'{y"}' (<-'y')
+		'{z<}' (<-'z')
+		'' (next)
+	)
+)
+
+define lower_case as repeat (
```

Comment:
I'm not sure if this is necessary: the stemmers usually received lower-cased input, no? And unicode-aware case folding generally does the right thing with Slovak.